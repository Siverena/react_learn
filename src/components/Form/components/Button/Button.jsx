import style from './button.module.scss';
export const Button = ({ label, disabled = false }) => (
  <button
    disabled={disabled}
    data-testid="submitbutton"
    className={style['button']}
  >
    {label}
  </button>
);
