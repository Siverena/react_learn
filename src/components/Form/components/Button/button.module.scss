.button{
        width: 80px;
        margin-left: 10px;
        cursor: pointer;
        border: 1px solid #ff6a6a;
        color: #f26376;
        background-color: transparent;
        &:hover{
            background-color: #f16d7f;
            color: #fff;
            border: none;
        }
    }