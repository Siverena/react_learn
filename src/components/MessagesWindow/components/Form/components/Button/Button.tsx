import style from './button.module.scss';
import React from 'react';
import { FC } from 'react';

interface ButtonProps {
  label: string;
  disabled?: boolean;
  click?: () => void;
}

export const Button: FC<ButtonProps> = ({ label, disabled = false, click }) => (
  <button
    disabled={disabled}
    data-testid="submitbutton"
    className={style['button']}
    onClick={click}
  >
    {label}
  </button>
);
