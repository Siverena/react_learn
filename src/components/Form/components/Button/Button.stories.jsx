import React from 'react';
import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
};

const Template = () => <Button />;

export const NormalButton = Template.bind({});
