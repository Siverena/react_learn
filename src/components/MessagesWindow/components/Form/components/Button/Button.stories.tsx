import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = () => (
  <Button label="отправить" />
);

export const NormalButton = Template.bind({}) as ComponentStory<typeof Button>;
