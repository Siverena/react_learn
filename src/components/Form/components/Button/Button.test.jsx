import React from 'react';
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';

import { Button } from './Button';

describe('Button', () => {
  it('render component', () => {
    render(<Button label="test" />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button label="test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('render component with text', () => {
    render(<Button label="test" />);
    expect(screen.getByText(/test/)).toBeInTheDocument();
  });
  it('button is disabled', () => {
    render(<Button label="test2" disabled />);
    expect(screen.getByText(/test2/)).toBeDisabled();
  });
});
