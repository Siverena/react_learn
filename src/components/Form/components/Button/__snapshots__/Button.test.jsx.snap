// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    data-testid="submitbutton"
  >
    test
  </button>
</DocumentFragment>
`;
